File: algorithms/088.Merge_Sorted_Array/Merge_Sorted_Array.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int ia = m - 1, ib = n - 1, icur = m + n - 1;
        while(ia >= 0 && ib >= 0)
            nums1[icur--] = nums1[ia] > nums2[ib] ? nums1[ia--] : nums2[ib--];
        while(ib >=0)
           nums1[icur--] = nums2[ib--];
    }
};

int main() {
    int a[100], b[100];
    int n, m;
    Solution s;
    while (cin >> n >> m) {
        for (int i = 0; i < n; i++)
            scanf("%d", &a[i]);
        for (int i = 0; i < m; i++)
            scanf("%d", &b[i]);
        s.merge(a, n, b, m);
        for (int i = 0; i < n + m; i++)
            printf("%d ", a[i]);
        puts("");
    }
    return 0;
}

File: algorithms/088.Merge_Sorted_Array/Merge_Sorted_Array.py
class Solution:
    #@param A: sorted integer array A which has m elements,
    #          but size of A is m+n
    #@param B: sorted integer array B which has n elements
    #@return: void
    def merge(self, nums1, m, nums2, n):
        # for i in range(n):
        #     A[i+m] = B[i]
        # A.sort()

    # def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i, j, k = m-1, n-1, m+n-1
        while i >= 0 and j >=0:
            if nums1[i] > nums2[j]:
                nums1[k] = nums1[i]
                k = k-1
                i = i-1
            else:
                nums1[k] = nums2[j]
                k = k-1
                j = j-1
        if j > 0:
            nums1[:n] = nums2[:n]


a = [1,2,3,0,0,0]
m = 3
b = [2,5,6]
n = 3

Solution().merge(a, m, b, n)

print(a)